torch/csrc/jit/fusion_compiler.h

```diff
@@ -125,14 +116,14 @@ struct FusionCompiler {
 
   // ignores types in graph, and uses specific contiguity annotations
   std::shared_ptr<CompiledFusionFunction> getOrCompile(AnnotatedGraph & agraph);
-  // uses type annotations in graph to create Annotated graph
-  std::shared_ptr<CompiledFusionFunction> getOrCompile(Graph & graph);
+  // uses type annotations in fusion_group to create Annotated graph
+  std::shared_ptr<CompiledFusionFunction> getOrCompile(Node * fusion_group);
 
   // debugging function that lets you do everything from compilation to execution
   // in one step.
   // this should not be used in the hot path of execution because it has to serialize
   // the graph each time
-  void debugLaunchGraph(Graph & graph, at::ArrayRef<at::Tensor> inputs, at::ArrayRef<at::Tensor> outputs);
+  void debugLaunchGraph(Graph & graph, bool is_cuda, at::ArrayRef<at::Tensor> inputs, at::ArrayRef<at::Tensor> outputs);
```

Comment:
isn't `is_cuda` a property of the graph? why is an argument needed?

Reply:
It's a part of the `kFusionGroup` Node  but not part of the Graph that is taken by the fuser
The fuser itself doesn't require you have types/devices attached for debugging and for keeping the potential open for it to be used directly from C++ for ATen tensors. If we gain more parameters, we can refactor into a thing that contains all the meta-data needed.